.github/workflows/cron-nightly-rust.yml

```
@@ -0,0 +1,57 @@
+name: Update nightly Rust
+on:
+  schedule:
+    - cron: "29 17 1 * *" # At 17:29 on day-of-month 1.
+  workflow_dispatch: # allows manual triggering
+env:
+  CARGO_TERM_COLOR: always
+jobs:
+  format:
+    name: Update nightly Rustc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: dtolnay/rust-toolchain@nightly
+      - name: Update rust-toolchain.toml to use latest nightly
+        run: |
+          set -x
```

Should fail on error as a best-practice.

```suggestion
          set -xe
```